### ISSUE TITLE
Unexpected replacement of :counter in resources→title/name

### ISSUE THREAD

**Issue by kaushalmodi** [gohugoio/hugo#4334]
Hello,

Details of this issue got buried in this [HugoDocs issue discussion][1]. I believe this is something that needs to be fixed, so cross-posting the issue details here.

## Problem

If a Resource file is "fetched" by one set of `resources` → `src` rule, and then again using a different `resources` → `src` rule later, `:counter` value substituted in `name` (probably in `title` too) stays stuck at 1.

The above problem statement might not make sense at first read, so please review the below details.

### Works

This scenario works as expected i.e. the `:counter` is doing expected substitutions in the Resource Name.

Here I have only:

```toml
[[resources]]
  src = "**.pdf"
  name = "pdf-file-:counter"
```

- [Test file "Resource Counters A" Hugo output][2] (*Markdown source linked within that test page*)
  - Scroll down to **Resources (Debug)** section on that page for `.Resources` debug info.

Counter in `name` increments fine.

### Does not work

In this scenario, the Resource that is "fetched" twice i.e. photo_specs.pdf has its counter stuck at 1.

Here I have:

```toml
[[resources]]
  src = "documents/photo_specs.pdf"
  title = "Photo Specifications"
[[resources]]
  src = "**.pdf"
  name = "pdf-file-:counter"
```

- [Test file "Resource Counters C" Hugo output][3] (*Markdown source linked within that test page*)
  - Scroll down to **Resources (Debug)** section on that page for `.Resources` debug info.

The difference from the "Works" case is that, here I first set the `title` for `"documents/photo_specs.pdf"`. 

The behavior is strange though.. it could be understood if the `name` rule had **no** effect on the `Name` of `"documents/photo_specs.pdf"`.. assuming that Name was set implicitly in the first rule. The problem is that the `Name` does get *almost* set as per the `name` rule but the count is always stuck at 1.

*In summary, if a resource is fetched by an `src` filter earlier, the `:counter` messes up if set in a later rule.*

Or, to put it another way, it seems like `:counter` can be used only in the set of rules that first "fetches" a resources.

*I hope all that made sense.. It was a bit complicated to explain the issue.*

---

### Other

*PS: Is it true that `:counter` is designed to work only in `title` and `name`, but not in `params`?*

[1]: https://github.com/gohugoio/hugoDocs/issues/308#issuecomment-360844191
[2]: https://hugo-sandbox.netlify.com/posts/resources-counters-a/
[3]: https://hugo-sandbox.netlify.com/posts/resources-counters-c/

**Comment by bep** [comment]
>PS: Is it true that :counter is designed to work only in title and name, but not in params?

Yes. But for the other questions, use the forum.

**Comment by kaushalmodi** [comment]
Alright, I'll post on the forums, but I've done a lot of testing.. the results don't seem right.
